relay-server/src/endpoints/healthcheck.rs

```diff
@@ -35,14 +35,14 @@ impl HealthcheckResponse {
 fn healthcheck_impl(message: IsHealthy) -> ResponseFuture<HttpResponse, Error> {
     let fut = async move {
         let addr = Healthcheck::from_registry();
-        addr.send(message.into()).await
+        addr.send(message).await
     };
 
     Box::new(
         fut.boxed_local()
             .compat()
             .map_err(|_| ())
-            .and_then(move |is_healthy| {
+            .and_then(move |is_healthy: bool| {
```

Comment:
Is the compiler not able to infer that? That would be problematic, and I think we should revise the signatures then.

Reply:
This is only because of this old code style, async/await code can infer this just fine.  I replaced this logic in https://github.com/getsentry/relay/pull/1405/commits/d68902ccfaded08b28dc90c3ed5295b5bed1ca94.  Since this is a logic change (or rather, it shouldn't be!) please review carefully.

Reply:
(yes, that code still hints because of the `true` in there.  but even if you don't it can infer)

Reply:
never mind, inference works just fine.  reverted.